main.cc

```diff
@@ -527,7 +498,7 @@ main(int argc, char* argv[])
 
       break;
     case 'S':
-      switch (argv[argn][2]) {
+      switch (qargs.at(argn).size() > 2 ? qargs.at(argn)[2].toLatin1() : '\0') {
```

Comment:
It's 2008. :-) Please just remove this entire #if 0 block and the case associated with it.